packages/ccd-js-gen/src/lib.ts

```diff
@@ -282,379 +418,691 @@ This function ensures the smart contract module is deployed on chain.
+                },
+                {
+                    name: transactionMetadataId,
+                    type: 'SDK.ContractTransactionMetadata',
+                },
+                ...(receiveParameter === undefined
+                    ? []
+                    : [
+                          {
+                              name: parameterId,
+                              type: receiveParameterTypeId,
+                          },
+                      ]),
+                {
+                    name: signerId,
+                    type: 'SDK.AccountSigner',
+                },
+            ],
+            returnType: 'Promise<SDK.TransactionHash.Type>',
+        })
+        .setBodyText(
+            [
+                `return ${contractClientId}.${genericContractId}.createAndSendUpdateTransaction(`,
+                `    SDK.EntrypointName.fromStringUnchecked('${entrypointName}'),`,
+                '    SDK.Parameter.toBuffer,',
+                `    ${transactionMetadataId},`,
+                ...(receiveParameter === undefined
+                    ? []
+                    : [`    ${createReceiveParameterFnId}(${parameterId}),`]),
+                `    ${signerId}`,
+                ');',
+            ].join('\n')
+        );
+
+    contractSourceFile
+        .addFunction({
+            docs: [
+                [
+                    `Dry-run an update-contract transaction to the '${entrypointName}' entrypoint of the '${contractName}' contract.`,
+                    `@param {${contractClientType}} ${contractClientId} The client for a '${contractName}' smart contract instance on chain.`,
+                    `@param {SDK.ContractAddress.Type | SDK.AccountAddress.Type} ${invokerId} - The address of the account or contract which is invoking this transaction.`,
+                    ...(receiveParameter === undefined
+                        ? []
+                        : [
+                              `@param {${receiveParameterTypeId}} ${parameterId} - Parameter to provide the smart contract entrypoint as part of the transaction.`,
+                          ]),
+                    `@param {SDK.BlockHash.Type} [${blockHashId}] - Optional block hash allowing for dry-running the transaction at the end of a specific block.`,
+                    '@throws {SDK.RpcError} If failing to communicate with the concordium node or if any of the checks fails.',
+                    '@returns {SDK.InvokeContractResult} The result of invoking the smart contract instance.',
+                ].join('\n'),
+            ],
+            isExported: true,
+            name: `dryRun${toPascalCase(entrypointName)}`,
+            parameters: [
+                {
+                    name: contractClientId,
+                    type: contractClientType,
+                },
+                {
+                    name: invokerId,
+                    type: 'SDK.ContractAddress.Type | SDK.AccountAddress.Type',
+                },
```

Comment:
This should probably be something similar to the metadata for sending transactions, since currently we don't provide a way to specify energy, amount and expiry, which could be relevant when dry-running.

Reply:
It is now possible to provide amount and energy.